### ISSUE TITLE
The Role of Randomization to Address Confounding Variables in Machine Learning

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#450]
The Role of Randomization to Address Confounding Variables in Machine Learning<br>
<br>
https://ift.tt/2utXHez<br>
<br>
Jason Brownlee<br>
<br>
<p>A large part of applied machine learning is about running controlled experiments to discover what algorithm or algorithm configuration to use on a predictive modeling problem.</p>
<p>A challenge is that there are aspects of the problem and the algorithm called confounding variables that cannot be controlled (held constant) and must be controlled-for. An example is the use of randomness in a learning algorithm, such as random initialization or random choices during learning.</p>
<p>The solution is to use randomness in a way that has become a standard in applied machine learning. We can learn more about the rationale for using randomness in controlled experiments by looking briefly at why randomness is used to manage confounding variables in medicine through the use of randomized clinical trials.</p>
<p>In this post, you will discover confounding variables and how we can address them using the tool of randomization.</p>
<p>After reading this post, you will know:</p>
<ul>
<li>Confounding variables correlated with the independent and dependent variable confuse the effects and impact the results of experiments.</li>
<li>Applied machine learning is concerned with controlled experiments that do suffer known confounding variables.</li>
<li>Randomization of experiments is the key to controlling for confounding variables in machine learning experiments.</li>
</ul>
<p>Let&rsquo;s get started.</p>
<div class="wp-caption">
<img src="https://machinelearningmastery.com/wp-content/uploads/2018/07/The-Role-of-Randomization-to-Address-Confounding-Variables-in-Machine-Learning.jpg" alt="The Role of Randomization to Address Confounding Variables in Machine Learning" width="640" class="wp-image-5801" height="394">
<p class="wp-caption-text">The Role of Randomization to Address Confounding Variables in Machine Learning<br>
Photo by <a href="https://www.flickr.com/photos/funkdooby/33051321096/">Funk Dooby</a>, some rights reserved.</p>
</div>
<h2>Overview</h2>
<p>This post is divided into four parts;l they are:</p>
<ol>
<li>Confounding Variables</li>
<li>Confounding in Machine Learning</li>
<li>Randomization of Experiments</li>
<li>Randomization in Machine Learning</li>
</ol>
<center>
<h3>Need help with Statistics for Machine Learning?</h3>
<p>Take my free 7-day email crash course now (with sample code).</p>
<p>Click to sign-up and also get a free PDF Ebook version of the course.</p>
<p><a href="https://machinelearningmastery.lpages.co/leadbox/142f75173f72a2%3A164f8be4f346dc/5750943224168448/" target="_blank">Download Your FREE Mini-Course</a></p>
</center>
<h2>Confounding Variables</h2>
<p>In an experiment, we are often interested in the effect of an independent variable on a dependent variable.</p>
<p>A confounding variable is a variable that confuses the relationship between the independent and the dependent variable.</p>
<blockquote>
<p>Confounding, sometimes referred to as confounding bias, is mostly described as a &lsquo;mixing&rsquo; or &lsquo;blurring&rsquo; of effects.</p>
</blockquote>
<p>&mdash; <a href="https://www.sciencedirect.com/science/article/pii/S0085253815529748">Confounding: What it is and how to deal with it</a>, 2008.</p>
<p>A confounding variable can influence the outcome of an experiment in many ways, such as:</p>
<ul>
<li>Invalid correlations.</li>
<li>Increasing variance.</li>
<li>Introducing a bias.</li>
</ul>
<p>A confounding variable may be known or unknown.</p>
<p>They are often characterized as having an association or correlation with both the independent and dependent variables.</p>
<p>Another characterization is that the confounding variable affects groups or observations differently.</p>
<blockquote>
<p>Confounding variables or confounders are often defined as the variables correlate (positively or negatively) with both the dependent variable and the independent variable. A Confounder is an extraneous variable whose presence affects the variables being studied so that the results do not reflect the actual relationship between the variables under study.</p>
</blockquote>
<p>&mdash; <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4017459/">How to control confounding effects by statistical analysis</a>, 2012.</p>
<p>The deeper difficulty of confounding variables is that it may not be obvious that they exist and are impacting results.</p>
<p>The effects of confounding variables are often not obvious or even identifiable unless they are specifically addressed in the design of the experiment or data collection method.</p>
<h2>Confounding in Machine Learning</h2>
<p>Confounding variables are traditionally a concern in applied statistics.</p>
<p>This is because in statistics we are often concerned with the effect of independent variables on dependent variables in data. Statistical methods are designed to discover and describe these relationships and confounding variables can essentially corrupt or invalidate discoveries.</p>
<p>Machine learning practitioners are typically interested in the skill of a predictive model and less concerned with the statistical correctness or interpretability of the model. As such, confounding variables are an important topic when it comes to data selection and preparation, but less important than they may be when developing descriptive statistical models.</p>
<p>Nevertheless, confounding variables are critically important in applied machine learning.</p>
<p>The evaluation of a machine learning model is an experiment with independent and dependent variables. As such, it is subject to confounding variables.</p>
<p>What may be surprising is that you already know this and that the gold-standard practices in applied machine learning address this. Therefore, being intimately aware of the confounding variables in machine learning experiments is required to understand the choice and interpretation of machine learning model evaluation.</p>
<p>Consider, what impacts the evaluation of a machine learning model, what are the independent variables?</p>
<p>Some examples include:</p>
<ul>
<li>The choice of data preparation schemes.</li>
<li>The choice of the samples in the training dataset.</li>
<li>The choice of the samples in the test dataset.</li>
<li>The choice of learning algorithm.</li>
<li>The choice of the initialization of the learning algorithm.</li>
<li>The choice of the configuration of the learning algorithm.</li>
</ul>
<p>Each of these choices will impact the dependent variable in a machine learning experiment, which is the chosen metric used to estimate the skill of the model when making predictions.</p>
<p>The evaluation of a machine learning model involves the design and execution of controlled experiments. A controlled experiment holds all elements constant except one element under study. The two most common types of controlled experiments in machine learning are:</p>
<ul>
<li>Controlled experiments to vary and evaluate learning algorithms.</li>
<li>Controlled experiments to vary and evaluate learning algorithm configurations.</li>
</ul>
<p>Nevertheless, there are confounding variables that the controlled experiments cannot hold constant. Specifically, there are sources of randomness, that if they were held constant would result in an invalid evaluation of the model. Three examples include:</p>
<ul>
<li>Randomness in the data sample.</li>
<li>Randomness in model initialization.</li>
<li>Randomness in the learning algorithm.</li>
</ul>
<p>For example, weights in a neural network are initialized to random values. Stochastic gradient descent randomizes the order of samples in an epoch to vary the types of updates performed. Random subsets of features are selected for each possible cut point in random forest. And many more examples.</p>
<p>Randomization in machine learning algorithms is not a bug; it is a feature intended to improve the performance of the model on average over classical deterministic methods.</p>
<blockquote>
<p>Randomness can be present in ML at many different levels, usually enhancing performance or alleviating problems and difficulties of classical methods.</p>
</blockquote>
<p>&mdash; <a href="https://www.researchgate.net/publication/316740225_Randomized_Machine_Learning_Approaches_Recent_Developments_and_Challenges">Randomized Machine Learning Approaches: Recent Developments and Challenges</a>, 2017.</p>
<p>These are confounding variables that we cannot hold constant. If they are held constant, the evaluation of the model will no longer provide insight into the generalizability of the result. We will know how well the model performs on a specific data sample or initialization of sequence of decisions during learning, but little idea on how the model will perform in general.</p>
<p>The way that we can handle confounding variables that we cannot control is by using randomization.</p>
<h2>Randomization of Experiments</h2>
<p>Randomization is a technique used in experimental design to give control over confounding variables that cannot (should not) be held constant.</p>
<p>For example, randomization is used in clinical experiments to control-for the biological differences between individual human beings when evaluating a treatment. It is the reason why a treatment must be evaluated on multiple individuals rather than on a single individual before the findings can be generalized.</p>
<blockquote>
<p>In randomization the random assignment of study subjects to exposure categories to breaking any links between exposure and confounders. This reduces potential for confounding by generating groups that are fairly comparable with respect to known and unknown confounding variables.</p>
</blockquote>
<p>&mdash; <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4017459/">How to control confounding effects by statistical analysis</a>, 2012.</p>
<p>Randomization is a simple tool in experimental design that allows the confounding variables to have their effect across a sample. It shifts the experiment from looking at an individual case to a collection of observations, where statistical tools are used to interpret the finding.</p>
<p>In medicine, randomization is the gold standard for evaluating a treatment and is called the randomized clinical trial. It is designed to remove not only the confounding effects of biological differences, but also the bias, such as the effect of the experimenter choosing the members of the treatment and non-treatment groups. You can imagine that a treatment would look very successful if the least-sick members of a cohort were chosen to be administered.</p>
<blockquote>
<p>An [Randomized clinical trial] is a special kind of cohort study, with the characteristic that patients are randomly assigned to the experimental group (with exposure) and the control group (without exposure). [&hellip;] Therefore, randomization helps to prevent selection by the clinician, and helps to establish groups that are equal with respect to relevant prognostic factors.</p>
</blockquote>
<p>&mdash; <a href="http://www.kidneyinternational-online.org/article/S0085-2538(15)52692-6/fulltext">The randomized clinical trial: An unbeatable standard in clinical research?</a>, 2007.</p>
<p>There are still confounding variables when using a randomized clinical trial. An example is the case where the experimenters know what treatment participants of the study are receiving. This can impact the way the experimenters interact with the participants, which in turn can impact the results of the experiment.</p>
<p>The answer is to use blinding where participants or experimenters do not know the treatment. Ideally, a double-blind experiment is adopted, ensuring that both participates and experimenters are unaware of their treatment.</p>
<blockquote>
<p>When feasible, it is strongly recommended that also after randomization, patients and clinicians do not know who receives the intervention and who does not. Studies may be single blind (either the patient or the clinician does not know who receives the treatment and who does not) or double blind (both the patient and the clinician do not know who receives the treatment).</p>
</blockquote>
<p>&mdash; <a href="http://www.kidneyinternational-online.org/article/S0085-2538(15)52692-6/fulltext">The randomized clinical trial: An unbeatable standard in clinical research?</a>, 2007.</p>
<p>Note, before we move on to look at the use of randomization in machine learning, consider that there are other approaches to managing the effect of confounding variables. <a href="https://en.wikipedia.org/wiki/Confounding#Decreasing_the_potential_for_confounding">Wikipedia has a good list here</a>.</p>
<h2>Randomization in Machine Learning</h2>
<p>Randomization is used in the evaluation of machine learning models to manage the uncontrollable confounding variables.</p>
<p>It is key to the standard ways described for evaluating machine learning models and the rationale for using methods such as data resampling and repeating experiments.</p>
<ul>
<li>Resampling methods are used to randomize the training and test datasets to help estimate the skill of training and evaluating models on random samples of data from the domain, rather than on a specific sample of data.</li>
<li>Evaluation experiments are repeated to help estimate the skill of the model with different random initialization and learning decisions, rather than on a single set of initial conditions and sequence of learning decisions.</li>
</ul>
<p>Randomization allows the machine learning practitioner to generalize a finding, to make it useful and applicable. It&rsquo;s the reason why careful design of the test harness and resampling method is important. It is the reason why we repeat the evaluation of a model and the reason we don&rsquo;t fix the seed on the pseudorandom number generator.</p>
<p>I talk more about these topics in the posts:</p>
<ul>
<li><a href="https://machinelearningmastery.com/randomness-in-machine-learning/">Embrace Randomness in Machine Learning</a></li>
<li><a href="https://machinelearningmastery.com/evaluate-skill-deep-learning-models/">How to Evaluate the Skill of Deep Learning Models</a></li>
</ul>
<h3>Should We Use a Blind Trial?</h3>
<p>When we take a closer look at why we use randomization, to control for confounding variables, it raises questions about the other confounders that we may not be controlling for.</p>
<p>For example, the machine learning practitioner knowing the skill of models prior to giving each model a chance to do its best via data preparation and hyperparameter tuning. Perhaps practitioners should blind themselves to remove the possibility of biasing the choice of final model.</p>
<p>The risk is that the practitioner that really likes artificial neural networks will &ldquo;<em>discover</em>&rdquo; a neural network configuration that outperforms other models.</p>
<p>At best it is a statistical fluke or violation of Occam&rsquo;s Razor for a parsimonious solution to a predictive modeling project; at worst, it is scientific fraud. The reason that clinicians aggressively removed this bias is people&rsquo;s lives were at risk. We may get to that point with machine learning algorithms, e.g. in cars.</p>
<p>In practice, today, I think this is good motivation for front-loading an experiment with a large and careful design and automating the execution and statistical interpretation of the results.</p>
<h2>Further Reading</h2>
<p>This section provides more resources on the topic if you are looking to go deeper.</p>
<ul>
<li><a href="https://en.wikipedia.org/wiki/Confounding">Confounding on Wikipedia</a></li>
<li><a href="https://en.wikipedia.org/wiki/Controlling_for_a_variable">Controlling for a variable on Wikipedia</a></li>
<li><a href="https://en.wikipedia.org/wiki/Randomized_controlled_trial">Randomized controlled trial on Wikipedia</a></li>
<li>
<a href="http://www.kidneyinternational-online.org/article/S0085-2538(15)52692-6/fulltext">The randomized clinical trial: An unbeatable standard in clinical research?</a>, 2007.</li>
<li>
<a href="https://www.sciencedirect.com/science/article/pii/S0085253815529748">Confounding: What it is and how to deal with it</a>, 2008.</li>
<li><a href="https://stats.stackexchange.com/questions/271694/confounding-variables-in-machine-learning-predictions">Confounding variables in machine learning predictions? on Cross Validated</a></li>
<li>
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4017459/">How to control confounding effects by statistical analysis</a>, 2012.</li>
<li>
<a href="https://www.researchgate.net/publication/316740225_Randomized_Machine_Learning_Approaches_Recent_Developments_and_Challenges">Randomized Machine Learning Approaches: Recent Developments and Challenges</a>, 2017.</li>
</ul>
<h2>Summary</h2>
<p>In this post, you discovered confounding variables and how we can address them using the tool of randomization.</p>
<p>Specifically, you learned:</p>
<ul>
<li>Confounding variables correlated with the independent and dependent variable and confuse the effects and impact the results of experiments.</li>
<li>Applied machine learning is concerned with controlled experiments that do suffer known confounding variables.</li>
<li>Randomization of experiments is the key to controlling for confounding variables in machine learning experiments.</li>
</ul>
<p>Do you have any questions?<br>
Ask your questions in the comments below and I will do my best to answer.</p>
<p>The post <a rel="nofollow" href="https://machinelearningmastery.com/confounding-variables-in-machine-learning/">The Role of Randomization to Address Confounding Variables in Machine Learning</a> appeared first on <a rel="nofollow" href="https://machinelearningmastery.com">Machine Learning Mastery</a>.</p>
<br>
<br>
<br>
<br>
AI|ML|DM|IR|PR<br>
<br>
via Machine Learning Mastery https://ift.tt/2if3vo8<br>
<br>
July 18, 2018 at 03:23AM
